### ISSUE TITLE
Error with pixelmatch ES import into an ES Playwright test

### ISSUE THREAD

**Issue by manudas** [mapbox/pixelmatch#143]
Hi there

I am trying to use Pixelmatch 6.0 (ES module pixelmatch) on a TS project

It shouldn't be a problem, however, when importing it in a Playwright test, I get this problem in the beginning of the test file (not even in the import statement):

`Error: require() of ES Module C:\Sources\RTCP.DepTMS\WebClient\PlanReview\BlueEthos.Web.UI\node_modules\pixelmatch\index.js from C:\Sources\RTCP.DepTMS\WebClient\PlanReview\BlueEthos.Web.UI\playwright\tests\integration\plan-review.spec.ts not supported.
Instead change the require of index.js in C:\Sources\RTCP.DepTMS\WebClient\PlanReview\BlueEthos.Web.UI\playwright\tests\integration\plan-review.spec.ts to a dynamic import() which is available in all CommonJS modules.playwright`

![image](https://github.com/user-attachments/assets/7a853182-08b8-405e-8392-e90b68c6f529)

Following the advice to use a dynamic import it works, but I don't understand why, as this is not a CommonJS module.

I've removed the @types/pixelmatch module as it was for a previous CommonJS version, but no luck.

What could be happening? this is a non sense.

Let me know if I can do some checks to find out what the problem is.
Thanks

**Comment by Med002rewired** [comment]
My device is hacked and its legal licence is charged

**Comment by manudas** [comment]
?

**Comment by Med002rewired** [comment]
![Devices - A list of devices (i.e.PDF](https://github.com/user-attachments/files/18029466/Devices.-.A.list.of.devices.i.e.PDF)

**Comment by manudas** [comment]
I don't know what are you talking about and I don't understand why are you sending random PDFs with no real content

**Comment by marcusNumminen** [comment]
I get similar problems, I get this:
(node:2923) ExperimentalWarning: CommonJS module /home/vsts/work/1/s/src/fixtures/testFixture.ts is loading ES Module /home/vsts/work/1/s/node_modules/pixelmatch/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

Any suggestions to get rid of this?

**Comment by wizardpisces** [comment]
Same issue, fallback to version 5  solve my problem

**Comment by wizardpisces** [comment]
After read changeLog of V6 " **Breaking: publish pixelmatch as a ES module. Use the older v5.3.0 if you need to support CommonJS environments.** "

**Comment by mourner** [comment]
Yep, pixelmatch no longer supports CommonJS environments — please use older versions if you need ancient legacy compatibility.
